src/cli/index.js

```diff
@@ -234,8 +235,104 @@ export function main({
     });
   };
 
-  //
   const runEventuallyWithNetwork = (mutexPort: ?string): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      const connectionOptions = {
+        port: +mutexPort || constants.SINGLE_INSTANCE_PORT,
+      };
+
+      function startServer() {
+        const clients = new Set();
+        const server = http.createServer(manager);
+
+        // The server must not prevent us from exiting
+        server.unref();
+
+        // No socket must timeout
+        server.timeout = 0;
+
+        // If we fail to setup the server, we ask the existing one for its name
+        server.on('error', () => {
+          reportServerName();
+        });
+
+        // If we succeed, keep track of all the connected sockets to delete them later
+        server.on('connection', socket => {
+          clients.add(socket);
+          socket.on('close', () => {
+            clients.delete(socket);
+          });
+        });
+
+          server.listen(mutexPort, () => {
+              console.log('ok', mutexPort);
+          // Don't forget to kill the sockets if we're being killed via signals
+          onDeath(killSockets);
+
+          // Also kill the sockets if we finish, whether it's a success or a failure
+          run().then(res => {
+              killSockets();
+              resolve(res);
+          }, err => {
+              killSockets();
+              reject(err);
+          });
+        });
+
+        function manager(request, response) {
+          response.writeHead(200);
+          response.write(config.cwd);
+          response.end();
+        }
+
+        function killSockets() {
+          for (const socket of clients) {
+            try {
+              socket.destroy();
+            } catch (err) {
+                // best effort
+            }
+          }
+        }
+      }
+
+      function reportServerName() {
+        const request = http.get({ port: mutexPort }, response => {
+          const buffers = [];
+
+          response.on('error', () => {
+            tryServer();
+          });
+
+          response.on('data', buffer => {
+            buffers.push(buffer);
+          });
+
+          response.on('end', () => {
+            reporter.warn(reporter.lang('waitingNamedInstance', Buffer.concat(buffers).toString()));
```

Comment:
I'd say this shouldn't be a warning.

Reply:
I agree, but it's a warning in the current code

Reply:
May be we can use the forced log here too?